src/plaid/containers/dataset.py

```diff
@@ -410,7 +424,52 @@ def get_field_names(
         return fields_names
 
     # -------------------------------------------------------------------------#
-    def add_tabular_scalars(self, tabular: np.ndarray, names: list[str] = None) -> None:
+    def get_all_features_identifiers(
+        self, ids: Optional[list[int]] = None
+    ) -> list[FeatureIdentifier]:
+        """Get all features identifiers from the dataset.
+
+        Args:
+            ids (list[int], optional): Sample id from which returning feature identifiers. If None, take all samples. Defaults to None.
+
+        Returns:
+            list[FeatureIdentifier]: A list of dictionaries containing the identifiers of all features in the dataset.
+        """
+        if ids is not None and len(set(ids)) != len(ids):
+            logger.warning("Provided ids are not unique")
+
+        all_features_identifiers = []
+        for sample in self.get_samples(ids, as_list=True):
+            features_identifiers = sample.get_all_features_identifiers()
+            for feat_id in features_identifiers:
+                if feat_id not in all_features_identifiers:
+                    all_features_identifiers.append(feat_id)
+        all_features_identifiers
+        return all_features_identifiers
+
+    def get_all_features_identifiers_by_type(
+        self, feature_type: str, ids: list[int] = None
```

Comment:
Given that we introduced `FeatureType` but here the function expect a string for something called `feature_type`, I think we should be more specific to help the next developper stepping on this: `feature_type: Literal["scalar", "field", "time_series", ...]` ?

In a future :recycle: , we could see if it would be appropriate the introduce an `Enum` for `FeatureType` instead of a `Union[Scalar, Field, ...]`